src/txn/txn_ckpt.c

```diff
@@ -1100,8 +1100,10 @@ __wt_checkpoint_close(WT_SESSION_IMPL *session, int final, int force)
 	bulk = F_ISSET(btree, WT_BTREE_BULK) ? 1 : 0;
 
 	/* Handle forced discard (when dropping a file). */
-	if (force)
-		return (__wt_cache_op(session, NULL, WT_SYNC_DISCARD_FORCE));
+	if (force) {
+		F_SET(session->dhandle, WT_DHANDLE_DEAD);
```

Comment:
@michaelcahill, I'd be inclined to move this test & set into the caller, it feels odd to me we're setting a handle flag here.

And: the subsequent case:
```
/*
 * If closing an unmodified file, check that no update is required
 * for active readers.
 */
if (!btree->modified && !bulk) {
        __wt_txn_update_oldest(session);
        return (__wt_txn_visible_all(session, btree->rec_max_txn) ?
            __wt_cache_op(session, NULL, WT_SYNC_DISCARD) : EBUSY);
}
```
could we set `WT_DHANDLE_DEAD` instead of calling `__wt_cache_op()`?